x-pack/plugins/apm/public/components/app/ErrorGroupDetails/DetailView/StickyErrorProperties.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { Fragment } from 'react';
+import {
+  ERROR_EXC_HANDLED,
+  HTTP_REQUEST_METHOD,
+  TRANSACTION_ID,
+  URL_FULL,
+  USER_ID
+} from 'x-pack/plugins/apm/common/elasticsearch_fieldnames';
+import { NOT_AVAILABLE_LABEL } from 'x-pack/plugins/apm/common/i18n';
+import { idx } from 'x-pack/plugins/apm/common/idx';
+import { APMError } from 'x-pack/plugins/apm/typings/es_schemas/Error';
+import { Transaction } from 'x-pack/plugins/apm/typings/es_schemas/Transaction';
+import { KibanaLink } from '../../../shared/Links/KibanaLink';
+import { legacyEncodeURIComponent } from '../../../shared/Links/url_helpers';
+import { StickyProperties } from '../../../shared/StickyProperties';
+
+interface Props {
+  error: APMError;
+  transaction: Transaction | undefined;
```

Comment:
Is this different from `transaction?: Transaction;` ?

Reply:
Yes, it is to force consumers to provide a transaction. The transaction might be `undefined` but the prop must be supplied.
I don't have strong opinions on this here but felt it made more sense.